File: ifa-portal-frontend/src/common/model/RequestModel.js
import store from '@/store'

// 共通リクエストBodyモデル。
export class RequestModel {
  // コンストラクタ。
  constructor(data) {
    this.frameworkSessionId = store.getters.user_sessionId
    this.authToken = store.getters.user_token
    this.postBody = data
  }
}

File: ifa-portal-backend/ifa-portal-common/src/main/java/com/sbisec/helios/ap/common/model/RequestModel.java
package com.sbisec.helios.ap.common.model;

import lombok.Data;

@Data
public class RequestModel<PostBody> {
	private String frameworkSessionId;
	private String authToken;
	
	private PostBody postBody;
	
	@SuppressWarnings("unchecked")
	public RequestModel(String frameworkSessionId, String authToken, Object object) {
		this.frameworkSessionId = frameworkSessionId;
		this.authToken = authToken;
		this.postBody = (PostBody) object;
	}
}
